salesforce/params.go

```diff
@@ -0,0 +1,69 @@
+package salesforce
+
+import (
+	"net/http"
+
+	"golang.org/x/oauth2"
+)
+
+type Option func(params *sfParams)
+
+func WithClient(client *http.Client) Option {
+	return func(params *sfParams) {
+		params.client = client
+	}
+}
+
+func WithOAuthToken(token *oauth2.Token) Option {
+	return func(params *sfParams) {
+		params.token = token
+	}
+}
+
+func WithOAuthConfig(config *oauth2.Config) Option {
+	return func(params *sfParams) {
+		params.config = config
+	}
+}
+
+func WithTokenSource(tokenSource oauth2.TokenSource) Option {
+	return func(params *sfParams) {
+		params.tokenSource = tokenSource
+	}
+}
+
+func WithWorkspace(workspaceRef string) Option {
+	return func(params *sfParams) {
+		params.workspaceRef = workspaceRef
+	}
+}
+
+type sfParams struct {
+	client       *http.Client
+	workspaceRef string
```

Comment:
Since these params look to be customizable for each connector, we might as well use Salesforce-specific terminology here. Salesforce calls this `subdomain`, I originally thought these parameters would be the same across all connectors, but I like this approach better.

Reply:
Ah good to know. Subdomain it is.